### PR TITLE
Add TPM SPI dumps

### PR DESCRIPTION
**Pull request by arturkow2000**

### CODE REVIEW

File: docs/tpm-spi-dump.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+Complete setup looks like this:
+
+![](/images/tpm_rpi_dslogic_connection.jpg)
+
+## Monitoring SPI bus
+
+In DSView open Device Options window, set operation mode to buffered (required
+for DSLogic Plus to operate at 100 MHz with 4 channels). Set voltage threshold
+to 2.0V and select max 4 channels.
```
[the remark this answers is not in the document]
Comment (arturkow2000) [reply docs/tpm-spi-dump.md:128]: @krystian-hebel Tried with default 1V and 2V, 1V didn't work correctly

File: docs/tpm-spi-dump.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+process is simple and is not covered here.
+
+## Preparing Raspberry PI
+
+Recent 32-bit RPI OS had a problem with non-working SPI chip select, we
+recommend using 64-bit version which worked properly. OS can be installed using
+[Raspberry Pi Imager](https://www.raspberrypi.com/software/). Please remember to
+configure user passwords before installing OS to avoid login problems.
+
+![/images/rpi_imager_cfg.png]
```
[the remark this answers is not in the document]
Comment (arturkow2000) [reply docs/tpm-spi-dump.md]: Fixed and added alt-text in all images (confirmed mkdocs can handle this)

File: docs/tpm-spi-dump.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+![](/images/dsview_proto_cfg.png)
+
+When configured properly DSView should show TPM registers we read or write, and
+raw data that goes to/from these registers.
+
+![](/images/dsview_trace_tpm_init.png)
+
+## Communication dumps
+
+`data` directory in this repository contains a few `*.dsl` files which can be
```
[the remark this answers is not in the document]
Comment (arturkow2000) [reply docs/tpm-spi-dump.md]: Done

File: docs/tpm-spi-dump.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+enable decoding click on the `Decode` button from toolbar and search for the
+`SPI TPM` protocol. Select proper CLK, MISO, MOSI, CS#, and leave other settings
+at theirs defaults.
+
+![](/images/dsview_proto_cfg.png)
+
+When configured properly DSView should show TPM registers we read or write, and
+raw data that goes to/from these registers.
+
+![](/images/dsview_trace_tpm_init.png)
```
[the remark this answers is not in the document]
Comment (arturkow2000) [reply docs/tpm-spi-dump.md]: Yes, those are saved in DSL files. I added signal names to all DSL files and updated this image

**Review by krystian-hebel — APPROVED** [review]

**Pull request merged**